Description: Adjust tests for ipywidgets 6
 Adjust some more tests to the argument order used with ipywidgets 6.
Author: Martin von Gagern <Martin.vGagern@gmx.net>

--- a/sage/src/sage/dynamics/complex_dynamics/mandel_julia.py
+++ b/sage/src/sage/dynamics/complex_dynamics/mandel_julia.py
@@ -599,7 +599,7 @@
     the form ``f = z^2 + c``)::
 
         sage: julia_plot(interact=True)
-        interactive(children=(FloatSlider(value=-1.0, description='Real c'...
+        interactive(children=(FloatSlider(value=-1.0,... description='Real c'...
 
         ::
 
--- a/sage/src/sage/interacts/library.py
+++ b/sage/src/sage/interacts/library.py
@@ -14,8 +14,8 @@
     sage: interacts.calculus.taylor_polynomial()
     Interactive function <function taylor_polynomial at ...> with 3 widgets
       title: HTMLText(value='<h2>Taylor polynomial</h2>')
-      f: EvalText(value='e^(-x)*sin(x)', description='$f(x)=$', layout=Layout(max_width='81em'))
-      order: SelectionSlider(description='order', options=(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12), value=1)
+      f: EvalText(value='e^(-x)*sin(x)', description='$f(x)=$')
+      order: SelectionSlider(value=1, options=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12], description='order')
 
 AUTHORS:
 
@@ -92,7 +92,7 @@
         ....:     print(n)
         sage: f()  # an interact appears if using the notebook, else code
         Interactive function <function f at ...> with 1 widget
-          n: IntSlider(value=5, description='n', max=15, min=-5)
+          n: IntSlider(value=5, min=-5, max=15, step=1, description='n')
     """
     @sage_wraps(f)
     def library_wrapper():
@@ -134,8 +134,8 @@
 
         sage: interacts.demo()
         Interactive function <function demo at ...> with 2 widgets
-          n: SelectionSlider(description='n', options=(0, 1, 2, 3, 4, 5, 6, 7, 8, 9), value=0)
-          m: SelectionSlider(description='m', options=(0, 1, 2, 3, 4, 5, 6, 7, 8, 9), value=0)
+          n: SelectionSlider(value=0, options=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9], description='n')
+          m: SelectionSlider(value=0, options=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9], description='m')
     """
     print(n + m)
 
@@ -162,8 +162,8 @@
         sage: interacts.calculus.taylor_polynomial()
         Interactive function <function taylor_polynomial at ...> with 3 widgets
           title: HTMLText(value='<h2>Taylor polynomial</h2>')
-          f: EvalText(value='e^(-x)*sin(x)', description='$f(x)=$', layout=Layout(max_width='81em'))
-          order: SelectionSlider(description='order', options=(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12), value=1)
+          f: EvalText(value='e^(-x)*sin(x)', description='$f(x)=$')
+          order: SelectionSlider(value=1, options=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12], description='order')
     """
     x0  = 0
     p   = plot(f,(x,-1,5), thickness=2)
@@ -205,12 +205,11 @@
         sage: interacts.calculus.definite_integral()
         Interactive function <function definite_integral at ...> with 6 widgets
           title: HTMLText(value='<h2>Definite integral</h2>')
-          f: EvalText(value='3*x', description='$f(x)=$', layout=Layout(max_width='81em'))
-          g: EvalText(value='x^2', description='$g(x)=$', layout=Layout(max_width='81em'))
-          interval: IntRangeSlider(value=(0, 3), description='Interval', max=10, min=-10)
-          x_range: IntRangeSlider(value=(0, 3), description='plot range (x)', max=10, min=-10)
-          selection: Dropdown(description='Select', index=2,
-                              options=('f', 'g', 'f and g', 'f - g'), value='f and g')
+          f: EvalText(value='3*x', description='$f(x)=$')
+          g: EvalText(value='x^2', description='$g(x)=$')
+          interval: IntRangeSlider(value=(0, 3), min=-10, max=10, step=1, description='Interval')
+          x_range: IntRangeSlider(value=(0, 3), min=-10, max=10, step=1, description='plot range (x)')
+          selection: Dropdown(value='f and g', options=['f', 'g', 'f and g', 'f - g'], description='Select')
     """
     x = SR.var('x')
     f = symbolic_expression(f).function(x)
@@ -289,9 +288,9 @@
         sage: interacts.calculus.function_derivative()
         Interactive function <function function_derivative at ...> with 4 widgets
           title: HTMLText(value='<h2>Derivative grapher</h2>')
-          function: EvalText(value='x^5-3*x^3+1', description='Function:', layout=Layout(max_width='81em'))
-          x_range: FloatRangeSlider(value=(-2.0, 2.0), description='Range (x)', max=15.0, min=-15.0)
-          y_range: FloatRangeSlider(value=(-8.0, 6.0), description='Range (y)', max=15.0, min=-15.0)
+          function: EvalText(value='x^5-3*x^3+1', description='Function:')
+          x_range: FloatRangeSlider(value=(-2.0, 2.0), min=-15.0, max=15.0, step=0.1, description='Range (x)')
+          y_range: FloatRangeSlider(value=(-8.0, 6.0), min=-15.0, max=15.0, step=0.1, description='Range (y)')
     """
     x = SR.var('x')
     f = symbolic_expression(function).function(x)
@@ -336,10 +335,10 @@
         sage: interacts.calculus.difference_quotient()
         Interactive function <function difference_quotient at ...> with 5 widgets
           title: HTMLText(value='<h2>Difference quotient</h2>')
-          f: EvalText(value='sin(x)', description='f(x)', layout=Layout(max_width='81em'))
-          interval: FloatRangeSlider(value=(0.0, 10.0), description='Range', max=10.0)
-          a: IntSlider(value=5, description='$a$', max=10)
-          x0: IntSlider(value=2, description='$x_0$ (start point)', max=10)
+          f: EvalText(value='sin(x)', description='f(x)')
+          interval: FloatRangeSlider(value=(0.0, 10.0), min=0.0, max=10.0, step=0.1, description='Range')
+          a: IntSlider(value=5, min=0, max=10, step=1, description='$a$')
+          x0: IntSlider(value=2, min=0, max=10, step=1, description='$x_0$ (start point)')
     """
     html('<h2>Difference Quotient</h2>')
     html('<div style="white-space: normal;">\
@@ -395,9 +394,9 @@
 
         sage: interacts.calculus.quadratic_equation()
         Interactive function <function quadratic_equation at ...> with 3 widgets
-          A: IntSlider(value=1, description='A', max=7, min=-7)
-          B: IntSlider(value=1, description='B', max=7, min=-7)
-          C: IntSlider(value=-2, description='C', max=7, min=-7)
+          A: IntSlider(value=1, min=-7, max=7, step=1, description='A')
+          B: IntSlider(value=1, min=-7, max=7, step=1, description='B')
+          C: IntSlider(value=-2, min=-7, max=7, step=1, description='C')
     """
     x = SR.var('x')
     f = symbolic_expression(A*x**2 + B*x + C).function(x)
@@ -456,9 +455,9 @@
 
         sage: interacts.geometry.trigonometric_properties_triangle()
         Interactive function <function trigonometric_properties_triangle at ...> with 3 widgets
-          a0: IntSlider(value=30, description='A', max=360)
-          a1: IntSlider(value=180, description='B', max=360)
-          a2: IntSlider(value=300, description='C', max=360)
+          a0: IntSlider(value=30, min=0, max=360, step=1, description='A')
+          a1: IntSlider(value=180, min=0, max=360, step=1, description='B')
+          a2: IntSlider(value=300, min=0, max=360, step=1, description='C')
     """
     import math
 
@@ -535,8 +534,8 @@
 
         sage: interacts.geometry.unit_circle()
         Interactive function <function unit_circle at ...> with 2 widgets
-          function: Dropdown(description='function', options=(('sin(x)', 0), ('cos(x)', 1), ('tan(x)', 2)), value=0)
-          x: TransformFloatSlider(value=0.0, description='x', max=6.283185307179586, step=0.015707963267948967)
+          function: Dropdown(value=0, options=[('sin(x)', 0), ('cos(x)', 1), ('tan(x)', 2)], description='function')
+          x: TransformFloatSlider(value=0.0, min=0.0, max=6.283185307179586, step=0.015707963267948967, description='x')
     """
     xy = (cos(x), sin(x))
     t = SR.var('t')
@@ -625,9 +624,9 @@
         sage: interacts.geometry.special_points()
         Interactive function <function special_points at ...> with 10 widgets
           title: HTMLText(value='<h2>Special points in triangle</h2>')
-          a0: IntSlider(value=30, description='A', max=360)
-          a1: IntSlider(value=180, description='B', max=360)
-          a2: IntSlider(value=300, description='C', max=360)
+          a0: IntSlider(value=30, min=0, max=360, step=1, description='A')
+          a1: IntSlider(value=180, min=0, max=360, step=1, description='B')
+          a2: IntSlider(value=300, min=0, max=360, step=1, description='C')
           show_median: Checkbox(value=False, description='Medians')
           show_pb: Checkbox(value=False, description='Perpendicular Bisectors')
           show_alt: Checkbox(value=False, description='Altitudes')
@@ -792,8 +791,8 @@
 
         sage: interacts.statistics.coin()
         Interactive function <function coin at ...> with 2 widgets
-          n: IntSlider(value=1000, description='Number of Tosses', max=10000, min=2, step=100)
-          interval: IntRangeSlider(value=(0, 0), description='Plotting range (y)', max=1)
+          n: IntSlider(value=1000, min=2, max=10000, step=100, description='Number of Tosses')
+          interval: IntRangeSlider(value=(0, 0), min=0, max=1, step=1, description='Plotting range (y)')
     """
     from random import random
     c = []
@@ -832,10 +831,10 @@
         sage: interacts.calculus.secant_method()
         Interactive function <function secant_method at ...> with 5 widgets
           title: HTMLText(value='<h2>Secant method for numerical root finding</h2>')
-          f: EvalText(value='x^2-2', description='f(x)', layout=Layout(max_width='81em'))
-          interval: IntRangeSlider(value=(0, 4), description='range', max=5, min=-5)
-          d: IntSlider(value=3, description='10^-d precision', max=16, min=1)
-          maxn: IntSlider(value=10, description='max iterations', max=15)
+          f: EvalText(value='x^2-2', description='f(x)')
+          interval: IntRangeSlider(value=(0, 4), min=-5, max=5, step=1, description='range')
+          d: IntSlider(value=3, min=1, max=16, step=1, description='10^-d precision')
+          maxn: IntSlider(value=10, min=0, max=15, step=1, description='max iterations')
     """
     def _bisection_method(f, a, b, maxn, eps):
         intervals = [(a,b)]
@@ -910,10 +909,10 @@
         sage: interacts.calculus.secant_method()
         Interactive function <function secant_method at ...> with 5 widgets
           title: HTMLText(value='<h2>Secant method for numerical root finding</h2>')
-          f: EvalText(value='x^2-2', description='f(x)', layout=Layout(max_width='81em'))
-          interval: IntRangeSlider(value=(0, 4), description='range', max=5, min=-5)
-          d: IntSlider(value=3, description='10^-d precision', max=16, min=1)
-          maxn: IntSlider(value=10, description='max iterations', max=15)
+          f: EvalText(value='x^2-2', description='f(x)')
+          interval: IntRangeSlider(value=(0, 4), min=-5, max=5, step=1, description='range')
+          d: IntSlider(value=3, min=1, max=16, step=1, description='10^-d precision')
+          maxn: IntSlider(value=10, min=0, max=15, step=1, description='max iterations')
     """
     def _secant_method(f, a, b, maxn, h):
         intervals = [(a,b)]
@@ -981,11 +980,11 @@
         sage: interacts.calculus.newton_method()
         Interactive function <function newton_method at ...> with 7 widgets
           title: HTMLText(value='<h2>Newton method</h2>')
-          f: EvalText(value='x^2 - 2', description='f', layout=Layout(max_width='81em'))
-          c: IntSlider(value=6, description='Start ($x$)', max=10, min=-10)
-          d: IntSlider(value=3, description='$10^{-d}$ precision', max=16, min=1)
-          maxn: IntSlider(value=10, description='max iterations', max=15)
-          interval: IntRangeSlider(value=(0, 6), description='Interval', max=10, min=-10)
+          f: EvalText(value='x^2 - 2', description='f')
+          c: IntSlider(value=6, min=-10, max=10, step=1, description='Start ($x$)')
+          d: IntSlider(value=3, min=1, max=16, step=1, description='$10^{-d}$ precision')
+          maxn: IntSlider(value=10, min=0, max=15, step=1, description='max iterations')
+          interval: IntRangeSlider(value=(0, 6), min=-10, max=10, step=1, description='Interval')
           list_steps: Checkbox(value=False, description='List steps')
     """
     def _newton_method(f, c, maxn, h):
@@ -1056,12 +1055,12 @@
         sage: interacts.calculus.trapezoid_integration()
         Interactive function <function trapezoid_integration at ...> with 7 widgets
           title: HTMLText(value='<h2>Trapezoid integration</h2>')
-          f: EvalText(value='x^2-5*x + 10', description='$f(x)=$', layout=Layout(max_width='81em'))
-          n: IntSlider(value=5, description='# divisions', min=1)
-          interval_input: ToggleButtons(description='Integration interval', options=('from slider', 'from keyboard'), value='from slider')
-          interval_s: IntRangeSlider(value=(0, 8), description='slider: ', max=10, min=-10)
-          interval_g: Grid(value=[[0, 8]], children=(Label(value='keyboard: '), VBox(children=(EvalText(value='0', layout=Layout(max_width='5em')),)), VBox(children=(EvalText(value='8', layout=Layout(max_width='5em')),))))
-          output_form: ToggleButtons(description='Computations form', options=('traditional', 'table', 'none'), value='traditional')
+          f: EvalText(value='x^2-5*x + 10', description='$f(x)=$')
+          n: IntSlider(value=5, min=1, max=100, step=1, description='# divisions')
+          interval_input: ToggleButtons(value='from slider', options=['from slider', 'from keyboard'], description='Integration interval')
+          interval_s: IntRangeSlider(value=(0, 8), min=-10, max=10, step=1, description='slider: ')
+          interval_g: Grid(value=[[0, 8]], description='keyboard: ', children=(Label(value='keyboard: '), VBox(children=(EvalText(value='0'),)), VBox(children=(EvalText(value='8'),))))
+          output_form: ToggleButtons(value='traditional', options=['traditional', 'table', 'none'], description='Computations form')
     """
     xs = []
     ys = []
@@ -1172,12 +1171,12 @@
         sage: interacts.calculus.simpson_integration()
         Interactive function <function simpson_integration at ...> with 7 widgets
           title: HTMLText(value='<h2>Simpson integration</h2>')
-          f: EvalText(value='x*sin(x)+x+1', description='$f(x)=$', layout=Layout(max_width='81em'))
-          n: IntSlider(value=6, description='# divisions', min=2, step=2)
-          interval_input: ToggleButtons(description='Integration interval', options=('from slider', 'from keyboard'), value='from slider')
-          interval_s: IntRangeSlider(value=(0, 10), description='slider: ', max=10, min=-10)
-          interval_g: Grid(value=[[0, 10]], children=(Label(value='keyboard: '), VBox(children=(EvalText(value='0', layout=Layout(max_width='5em')),)), VBox(children=(EvalText(value='10', layout=Layout(max_width='5em')),))))
-          output_form: ToggleButtons(description='Computations form', options=('traditional', 'table', 'none'), value='traditional')
+          f: EvalText(value='x*sin(x)+x+1', description='$f(x)=$')
+          n: IntSlider(value=6, min=2, max=100, step=2, description='# divisions')
+          interval_input: ToggleButtons(value='from slider', options=['from slider', 'from keyboard'], description='Integration interval')
+          interval_s: IntRangeSlider(value=(0, 10), min=-10, max=10, step=1, description='slider: ')
+          interval_g: Grid(value=[[0, 10]], description='keyboard: ', children=(Label(value='keyboard: '), VBox(children=(EvalText(value='0'),)), VBox(children=(EvalText(value='10'),))))
+          output_form: ToggleButtons(value='traditional', options=['traditional', 'table', 'none'], description='Computations form')
     """
     x = SR.var('x')
     f = symbolic_expression(f).function(x)
@@ -1303,12 +1302,12 @@
         sage: interacts.calculus.riemann_sum()
         Manual interactive function <function riemann_sum at ...> with 9 widgets
           title: HTMLText(value='<h2>Riemann integral with random sampling</h2>')
-          f: EvalText(value='x^2+1', description='$f(x)=$', layout=Layout(max_width='41em'))
-          n: IntSlider(value=5, description='# divisions', max=30, min=1)
+          f: EvalText(value='x^2+1', description='$f(x)=$')
+          n: IntSlider(value=5, min=1, max=30, step=1, description='# divisions')
           hr1: HTMLText(value='<hr>')
-          interval_input: ToggleButtons(description='Integration interval', options=('from slider', 'from keyboard'), value='from slider')
-          interval_s: IntRangeSlider(value=(0, 2), description='slider: ', max=10, min=-5)
-          interval_g: Grid(value=[[0, 2]], children=(Label(value='keyboard: '), VBox(children=(EvalText(value='0', layout=Layout(max_width='5em')),)), VBox(children=(EvalText(value='2', layout=Layout(max_width='5em')),))))
+          interval_input: ToggleButtons(value='from slider', options=['from slider', 'from keyboard'], description='Integration interval')
+          interval_s: IntRangeSlider(value=(0, 2), min=-5, max=10, step=1, description='slider: ')
+          interval_g: Grid(value=[[0, 2]], description='keyboard: ', children=(Label(value='keyboard: '), VBox(children=(EvalText(value='0'),)), VBox(children=(EvalText(value='2'),))))
           hr2: HTMLText(value='<hr>')
           list_table: Checkbox(value=False, description='List table')
 
@@ -1391,10 +1390,10 @@
         Interactive function <function function_tool at ...> with 7 widgets
           f: EvalText(value='sin(x)', description='f')
           g: EvalText(value='cos(x)', description='g')
-          xrange: IntRangeSlider(value=(0, 1), description='x-range', max=3, min=-3)
+          xrange: IntRangeSlider(value=(0, 1), min=-3, max=3, step=1, description='x-range')
           yrange: Text(value='auto', description='yrange')
-          a: IntSlider(value=1, description='a', max=3, min=-1)
-          action: ToggleButtons(description='h = ', options=('f', 'df/dx', 'int f', 'num f', 'den f', '1/f', 'finv', 'f+a', 'f-a', 'f*a', 'f/a', 'f^a', 'f(x+a)', 'f(x*a)', 'f+g', 'f-g', 'f*g', 'f/g', 'f(g)'), value='f')
+          a: IntSlider(value=1, min=-1, max=3, step=1, description='a')
+          action: ToggleButtons(value='f', options=['f', 'df/dx', 'int f', 'num f', 'den f', '1/f', 'finv', 'f+a', 'f-a', 'f*a', 'f/a', 'f^a', 'f(x+a)', 'f(x*a)', 'f+g', 'f-g', 'f*g', 'f/g', 'f(g)'], description='h = ')
           do_plot: Checkbox(value=True, description='Draw Plots')
     """
     x = SR.var('x')
@@ -1516,14 +1515,14 @@
 
         sage: interacts.fractals.julia()
         Interactive function <function julia at ...> with 8 widgets
-          expo: FloatSlider(value=2.0, description='expo', max=10.0, min=-10.0)
-          c_real: FloatSlider(value=0.5, description='real part const.', max=2.0, min=-2.0, step=0.01)
-          c_imag: FloatSlider(value=0.5, description='imag part const.', max=2.0, min=-2.0, step=0.01)
-          iterations: IntSlider(value=20, description='# iterations', min=1)
-          zoom_x: FloatRangeSlider(value=(-1.5, 1.5), description='Zoom X', max=2.0, min=-2.0, step=0.01)
-          zoom_y: FloatRangeSlider(value=(-1.5, 1.5), description='Zoom Y', max=2.0, min=-2.0, step=0.01)
-          plot_points: IntSlider(value=150, description='plot points', max=400, min=20, step=20)
-          dpi: IntSlider(value=80, description='dpi', max=200, min=20, step=10)
+          expo: FloatSlider(value=2.0, min=-10.0, max=10.0, step=0.1, description='expo')
+          c_real: FloatSlider(value=0.5, min=-2.0, max=2.0, step=0.01, description='real part const.')
+          c_imag: FloatSlider(value=0.5, min=-2.0, max=2.0, step=0.01, description='imag part const.')
+          iterations: IntSlider(value=20, min=1, max=100, step=1, description='# iterations')
+          zoom_x: FloatRangeSlider(value=(-1.5, 1.5), min=-2.0, max=2.0, step=0.01, description='Zoom X')
+          zoom_y: FloatRangeSlider(value=(-1.5, 1.5), min=-2.0, max=2.0, step=0.01, description='Zoom Y')
+          plot_points: IntSlider(value=150, min=20, max=400, step=20, description='plot points')
+          dpi: IntSlider(value=80, min=20, max=200, step=10, description='dpi')
     """
     z = SR.var('z')
     I = CDF.gen()
@@ -1565,12 +1564,12 @@
 
         sage: interacts.fractals.mandelbrot()
         Interactive function <function mandelbrot at ...> with 6 widgets
-          expo: FloatSlider(value=2.0, description='expo', max=10.0, min=-10.0)
-          iterations: IntSlider(value=20, description='# iterations', min=1)
-          zoom_x: FloatRangeSlider(value=(-2.0, 1.0), description='Zoom X', max=2.0, min=-2.0, step=0.01)
-          zoom_y: FloatRangeSlider(value=(-1.5, 1.5), description='Zoom Y', max=2.0, min=-2.0, step=0.01)
-          plot_points: IntSlider(value=150, description='plot points', max=400, min=20, step=20)
-          dpi: IntSlider(value=80, description='dpi', max=200, min=20, step=10)
+          expo: FloatSlider(value=2.0, min=-10.0, max=10.0, step=0.1, description='expo')
+          iterations: IntSlider(value=20, min=1, max=100, step=1, description='# iterations')
+          zoom_x: FloatRangeSlider(value=(-2.0, 1.0), min=-2.0, max=2.0, step=0.01, description='Zoom X')
+          zoom_y: FloatRangeSlider(value=(-1.5, 1.5), min=-2.0, max=2.0, step=0.01, description='Zoom Y')
+          plot_points: IntSlider(value=150, min=20, max=400, step=20, description='plot points')
+          dpi: IntSlider(value=80, min=20, max=200, step=10, description='dpi')
     """
     x, z, c = SR.var('x, z, c')
     f = symbolic_expression(z**expo + c).function(z, c)
@@ -1609,9 +1608,9 @@
 
         sage: interacts.fractals.cellular_automaton()
         Interactive function <function cellular_automaton at ...> with 3 widgets
-          N: IntSlider(value=100, description='Number of iterations', max=500, min=1)
-          rule_number: IntSlider(value=110, description='Rule number', max=255)
-          size: IntSlider(value=6, description='size of graphic', max=11, min=1)
+          N: IntSlider(value=100, min=1, max=500, step=1, description='Number of iterations')
+          rule_number: IntSlider(value=110, min=0, max=255, step=1, description='Rule number')
+          size: IntSlider(value=6, min=1, max=11, step=1, description='size of graphic')
     """
     from sage.rings.integer import Integer
     if not 0 <= rule_number <= 255:
@@ -1667,12 +1666,12 @@
 
         sage: sage.interacts.algebra.polar_prime_spiral()
         Interactive function <function polar_prime_spiral at ...> with 6 widgets
-          interval: IntRangeSlider(value=(1, 1000), description='range', max=4000, min=1, step=10)
+          interval: IntRangeSlider(value=(1, 1000), min=1, max=4000, step=10, description='range')
           show_factors: Checkbox(value=True, description='show_factors')
           highlight_primes: Checkbox(value=True, description='highlight_primes')
           show_curves: Checkbox(value=True, description='show_curves')
-          n: IntSlider(value=89, description='number $n$', max=200, min=1)
-          dpi: IntSlider(value=100, description='dpi', max=300, min=10, step=10)
+          n: IntSlider(value=89, min=1, max=200, step=1, description='number $n$')
+          dpi: IntSlider(value=100, min=10, max=300, step=10, description='dpi')
     """
     html('<h2>Polar Prime Spiral</h2> \
           <div style="white-space: normal;">\
--- a/sage/src/sage/interacts/test_jupyter.rst
+++ b/sage/src/sage/interacts/test_jupyter.rst
@@ -291,7 +291,7 @@
     ....:     print(parent(A))
     sage: test(matrix_test)
     Interactive function <function matrix_test at ...> with 1 widget
-      A: Grid(value=[[0, 1], [2, 3]], children=(Label(value=u'A'), VBox(children=(EvalText(value=u'0', layout=Layout(max_width=u'5em')), EvalText(value=u'2', layout=Layout(max_width=u'5em')))), VBox(children=(EvalText(value=u'1', layout=Layout(max_width=u'5em')), EvalText(value=u'3', layout=Layout(max_width=u'5em'))))))
+      A: Grid(value=[[0, 1], [2, 3]], description='A', children=(Label(value='A'), VBox(children=(EvalText(value='0'), EvalText(value='2'))), VBox(children=(EvalText(value='1'), EvalText(value='3')))))
     [0 1]
     [2 3]
     Full MatrixSpace of 2 by 2 dense matrices over Rational Field
--- a/sage/src/sage/repl/ipython_kernel/interact.py
+++ b/sage/src/sage/repl/ipython_kernel/interact.py
@@ -163,7 +163,7 @@
             EvalText(value='sin(x)')
             sage: from sage.plot.colors import Color
             sage: sage_interactive.widget_from_single_value(matrix([[1, 2], [3, 4]]))
-            Grid(value=[[1, 2], [3, 4]], children=(Label(value=''), VBox(children=(EvalText(value='1', layout=Layout(max_width='5em')), EvalText(value='3', layout=Layout(max_width='5em')))), VBox(children=(EvalText(value='2', layout=Layout(max_width='5em')), EvalText(value='4', layout=Layout(max_width='5em'))))))
+            Grid(value=[[1, 2], [3, 4]], children=(Label(value=''), VBox(children=(EvalText(value='1'), EvalText(value='3'))), VBox(children=(EvalText(value='2'), EvalText(value='4')))))
             sage: sage_interactive.widget_from_single_value(Color('cornflowerblue'))
             SageColorPicker(value='#6495ed')
         """
--- a/sage/src/sage/plot/plot3d/plot3d.py
+++ b/sage/src/sage/plot/plot3d/plot3d.py
@@ -1007,7 +1007,7 @@
         ....: def _(which_plot=[A,B,C,D,E]):
         ....:     show(which_plot)
         Interactive function <function _ at ...> with 1 widget
-          which_plot: Dropdown(description='which_plot', options=(Graphics3d Object, Graphics3d Object, Graphics3d Object, Graphics3d Object, Graphics3d Object), value=Graphics3d Object)
+          which_plot: Dropdown(value=Graphics3d Object, options=[Graphics3d Object, Graphics3d Object, Graphics3d Object, Graphics3d Object, Graphics3d Object], description='which_plot')
 
     Now plot a function::
 
@@ -1021,7 +1021,7 @@
         ....: def _(which_plot=[F, G, H, I, J]):
         ....:     show(which_plot)
         Interactive function <function _ at ...> with 1 widget
-          which_plot: Dropdown(description='which_plot', options=(Graphics3d Object, Graphics3d Object, Graphics3d Object, Graphics3d Object, Graphics3d Object), value=Graphics3d Object)
+          which_plot: Dropdown(value=Graphics3d Object, options=[Graphics3d Object, Graphics3d Object, Graphics3d Object, Graphics3d Object, Graphics3d Object], description='which_plot')
 
     TESTS:
 
